src/Contracts/RequestPaginator.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Contracts;
+
+use Countable;
+use GuzzleHttp\Promise\PromiseInterface;
+use Iterator;
+use ReturnTypeWillChange;
+
+/**
+ * @template TRequest of \Saloon\Contracts\Request
+ * @template TResponse of \Saloon\Contracts\Response
+ *
+ * @extends \Iterator<TResponse|\GuzzleHttp\Promise\PromiseInterface>
+ */
+interface RequestPaginator extends Countable, Iterator
+{
+    /**
+     * @param callable(int $pendingRequests): (int)|int $concurrency
+     * @param callable(TResponse $response, array-key $key, \GuzzleHttp\Promise\PromiseInterface $poolAggregate): (void)|null $responseHandler
+     * @param callable(mixed $reason, array-key $key, \GuzzleHttp\Promise\PromiseInterface $poolAggregate): (void)|null $exceptionHandler
+     */
+    public function pool(
+        callable|int $concurrency = 5,
+        ?callable $responseHandler = null,
+        ?callable $exceptionHandler = null,
+    ): Pool;
+
+    /**
+     * @return $this
+     */
+    public function async(bool $async = true): static;
+
+    /**
+     * @return bool
+     */
+    public function isAsync(): bool;
+
+    /**
+     * Makes the Paginator continue where it left off in an earlier loop, instead of rewinding/'resetting' when used in a new loop.
+     *
+     * Iterables are, by nature, rewinding ('resetting') whenever you use it in a new loop.
+     * I.e., if you go through 2 iterations in a loop, then break out of it, and iterate again in a new loop, it'll be rewound to the first element.
+     * This makes sure that the rewinding step is skipped.
+     *
+     * @param bool $ignoreRewinding
+     *
+     * @return $this
+     *
+     * @TODO Come up with a better name for this method.
+     */
+    public function ignoreRewinding(bool $ignoreRewinding = true): static;
+
+    /**
+     * Returns whether or not the Paginator will continue where it left off in a previous loop, or not.
+     *
+     * @return bool
+     *
+     * @see \Saloon\Contracts\RequestPaginator::ignoreRewinding()
+     *
+     * @TODO Come up with a better name for this method.
+     */
+    public function shouldRewind(): bool;
```

Comment:
I think `shouldRewind` is fine but you could call it `isRewindingEnabled`

Reply:
I re-used the method to check if we're on the last page, or if rewinding is enabled.
So will keep `shouldRewind()` for now.

I added a `rewindingEnabled()` method that explicitly returns the property.
I might rename it to `isRewindingEnabled()`, though.